connectors/sources/tests/test_confluence.py

```diff
@@ -161,7 +318,168 @@ def test_validate_configuration_for_ssl_enabled(patch_logger):
+
+    async_response = MockJsonResponse(RESPONSE_SPACE, 200)
+
+    # Execute
+    with mock.patch("aiohttp.ClientSession.get", return_value=async_response):
+        async for response in source.fetch_spaces():
+            assert response == EXPECTED_SPACE
+
+
+@pytest.mark.asyncio
+async def test_fetch_documents():
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    async_response = MockJsonResponse(RESPONSE_PAGE, 200)
+
+    # Execute
+    with mock.patch("aiohttp.ClientSession.get", return_value=async_response):
+        async for response, _ in source.fetch_documents(api_query=""):
+            assert response == EXPECTED_PAGE
+
+
+@pytest.mark.asyncio
+async def test_fetch_attachments():
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    async_response = MockJsonResponse(RESPONSE_ATTACHMENT, 200)
+
+    # Execute
+    with mock.patch("aiohttp.ClientSession.get", return_value=async_response):
+        async for response, _ in source.fetch_attachments(
+            content_id=1113,
+            parent_name="ES-scrum",
+            parent_space="DEMO",
+            parent_type="page",
+        ):
+            assert response == EXPECTED_ATTACHMENT
+
+
+@pytest.mark.asyncio
+async def test_download_attachment():
+    # Setup
+    source = create_source(ConfluenceDataSource)
+
+    async_response = MockObjectResponse()
+
+    # Execute
+    with mock.patch("aiohttp.ClientSession.get", return_value=async_response):
+        with mock.patch(
+            "aiohttp.StreamReader.iter_chunked",
+            return_value=AsyncIterator([bytes(RESPONSE_CONTENT, "utf-8")]),
+        ):
+            response = await source.download_attachment(
+                url="download/attachments/1113/demo.py?version=1&modificationDate=1672737890633&cacheVersion=1&api=v2",
+                attachment=EXPECTED_ATTACHMENT,
+                doit=True,
+            )
+            assert response == EXPECTED_CONTENT
+
+
+@pytest.mark.asyncio
+async def test_download_attachment_when_filesize_is_large():
+    """Tests the download attachments method for file size greater than max limit."""
+    # Setup
+    source = create_source(ConfluenceDataSource)
+
+    async_response = MockObjectResponse()
+
+    # Execute
+
+    with mock.patch("aiohttp.ClientSession.get", return_value=async_response):
+        with mock.patch(
+            "aiohttp.StreamReader.iter_chunked",
+            return_value=AsyncIterator([bytes(RESPONSE_CONTENT, "utf-8")]),
+        ):
+            attachment = copy(EXPECTED_ATTACHMENT)
+            attachment["size"] = 23000000
+
+            response = await source.download_attachment(
+                url="download/attachments/1113/demo.py?version=1&modificationDate=1672737890633&cacheVersion=1&api=v2",
+                attachment=attachment,
+                doit=True,
+            )
+            assert response is None
+
+
+@pytest.mark.asyncio
+async def test_download_attachment_for_unsupported_filetype():
```

Comment:
```suggestion
async def test_download_attachment_when_unsupported_filetype_used_then_fail_download_fails():
```

Same as above: What's happening? Does the download file or is it simply skipped?

Reply:
In this case too the content download is skipped, renaming test to `test_download_attachment_when_unsupported_filetype_used_then_fail_download_skips`